dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs

```diff
@@ -568,7 +581,9 @@ private static ChatResponseFormat GetJsonSchemaResponseFormat(Type formatObjectT
         var schema = KernelJsonSchemaBuilder.Build(type, configuration: s_jsonSchemaMapperConfiguration);
         var schemaBinaryData = BinaryData.FromString(schema.ToString());
 
-        return ChatResponseFormat.CreateJsonSchemaFormat(type.Name, schemaBinaryData, jsonSchemaIsStrict: true);
+        var typeName = StructuredOutputsJsonSchemaNameRegex().Replace(type.Name, string.Empty);
+
+        return ChatResponseFormat.CreateJsonSchemaFormat(typeName, schemaBinaryData, jsonSchemaIsStrict: true);
```

Comment:
Just wondering, does the type name even matter at all, e.g. can we not just always pass some static value like "SKType"?

Reply:
Good question! Based on my testing, JSON schema format **name** property is not used by the model to determine the nature of the data. For example, if I'm requesting a collection of animals, but my schema has name `birds`, the model result will include different animals, not just birds. On the other hand, JSON schema format **description** property is used by the model.

I couldn't find more information about **name** property in OpenAI documentation, but all of the examples of Structured Outputs functionality use the name that is related to actual format - `math_reasoning`, `get_weather`, `ui` and so on. I think there may be a case when JSON schema format is cached on OpenAI side by name or something like that. 

In all cases, I've decided to follow the same approach in our code as presented in OpenAI examples. I also checked OpenAI/SK Python SDKs, and it appeared they use the same approach, so we are also consistent here.

Reply:
Sounds good.  Better try and do something semantic in case they start using it. 😃